internal/core/src/common/Vector.h

```diff
@@ -58,18 +59,35 @@ class BaseVector {
 
 using VectorPtr = std::shared_ptr<BaseVector>;
 
+class SimpleVector : public BaseVector {
```

Comment:
why add this class?

Reply:
In custom functions like `StartsWithVarchar`, we need a base class for `ColumnVector`, `ConstantVector` (and `BitmapVector` in the future). So I add `SimpleVector` as their base class. `BaseVector` only has `size()` and `type` methods, which is unable to meet the requirements.

```cpp
void
StartsWithVarchar(const RowVector& args, FilterFunctionReturn& result) {
    Assert(args.childrens().size() == 2);
    auto strs = std::dynamic_pointer_cast<SimpleVector>(args.child(0));
    if (strs == nullptr) {
        PanicInfo(ExprInvalid, "invalid vector type");
    }
    Assert(strs->type() == DataType::VARCHAR ||
           strs->type() == DataType::STRING);
    auto prefixes = std::dynamic_pointer_cast<SimpleVector>(args.child(1));
    if (prefixes == nullptr) {
        PanicInfo(ExprInvalid, "invalid vector type");
    }
```